pkg/dependencies/factory.go

```diff
@@ -180,6 +180,7 @@ func (f *Factory) WithProviderFactory(clusterConfig *v1alpha1.Cluster) *Factory
 			VSphereGovcClient:         f.dependencies.Govc,
 			VSphereKubectlClient:      f.dependencies.Kubectl,
 			TinkerbellKubectlClient:   f.dependencies.Kubectl,
+			TinkerbellTinkClient:      f.dependencies.Tink,
```

Comment:
No need to make any changes to this PR about this, it is more for future work
Do y'all plan on adding more `executables` and/or other dependencies to tinkerbell?

Having this factory and the provider factory together makes a bit cumbersome to inject new dependencies (as you saw, you need to define them and pass them through 3 methods). I think this is totally ok for now, but we should look into unifying both if we want to add more of them

Let me know what you think 🙂

Reply:
Good point! I don't think we have any plans to add more dependencies to Tinkerbell. I would definitely keep this in mind if we plan in the future. Thanks!